js/ui/viewSelector.js

```diff
@@ -126,26 +128,116 @@ const ViewsDisplayLayout = new Lang.Class({
     Signals: { 'allocated-size-changed': { param_types: [GObject.TYPE_INT,
                                                          GObject.TYPE_INT] } },
 
-    _init: function(appDisplayActor) {
+    _init: function(entry, appDisplayActor, searchResultsActor) {
         this.parent();
 
+        this._entry = entry;
         this._appDisplayActor = appDisplayActor;
+        this._searchResultsActor = searchResultsActor;
+
+        this._entry.connect('style-changed', Lang.bind(this, this._onStyleChanged));
         this._appDisplayActor.connect('style-changed', Lang.bind(this, this._onStyleChanged));
+
+        this._heightAboveEntry = 0;
+        this.searchResultsTween = 0;
+        this._lowResolutionMode = false;
     },
 
     _onStyleChanged: function() {
         this.layout_changed();
     },
 
-    vfunc_allocate: function (actor, box, flags) {
-        let availWidth = box.x2 - box.x1;
-        let availHeight = box.y2 - box.y1;
+    _centeredHeightAbove: function (height, availHeight) {
+        return Math.floor(Math.max((availHeight - height) / 2, 0));
```

Comment:
No, it's my fault, and github does not make it any easier: I squashed by mistake some of the changes that should go into that commit in the previous one, leaving just a small subset of the changes required to add the search entry in this commit. I'll fix that